From: Michael Tokarev <mjt@tls.msk.ru>
Subject: force tevent to be standalone
Date: Tue, 26 Nov 2024 17:28:51 +0300
Forwarded: not-needed

Upstream ships contents of lib/tevent/ as a separate
source of tevent.  Since we build samba anyway, there's
no need to have separate tevent source package, it's
enough to build it during samba build.

Always build tevent as stand-alone library
(instead of being samba-private if in a subdir).

When building as a sub-library within samba, where
libreplace is a private library, libtevent.so will
have rpath pointing to the private samba dir.  Since
tevent actually does not use anything from libreplace,
just remove the dependency.

diff --git a/lib/tevent/wscript b/lib/tevent/wscript
index 8df1b40febf..99e64421da4 100644
--- a/lib/tevent/wscript
+++ b/lib/tevent/wscript
@@ -43,3 +43,3 @@ def configure(conf):
 
-    conf.env.standalone_tevent = conf.IN_LAUNCH_DIR()
+    conf.env.standalone_tevent = conf.IN_LAUNCH_DIR() or True
 
diff --git a/lib/tevent/wscript b/lib/tevent/wscript
index 8df1b40febf..34163127dd9 100644
--- a/lib/tevent/wscript
+++ b/lib/tevent/wscript
@@ -101,5 +101,5 @@ def build(bld):
 
     if not bld.CONFIG_SET('USING_SYSTEM_TEVENT'):
-        tevent_deps = 'replace talloc'
+        tevent_deps = 'talloc' # 'replace talloc'
         if bld.CONFIG_SET('HAVE_PTHREAD'):
             tevent_deps += ' pthread'
